tests/integration/helpers.py

```diff
@@ -139,6 +140,50 @@ def restart_unit(model_full_name: str, unit: str) -> None:
     )
 
 
+@retry(
+    stop=stop_after_attempt(3),
+    wait=wait_fixed(5),
+    retry_error_callback=lambda _: False,
+    retry=lambda x: x is False,
+)
+def dashboard_unavailable(host: str, https: bool = False) -> bool:
+    try:
+        # Normal IP address
+        socket.inet_aton(host)
+    except OSError:
+        socket.inet_pton(socket.AF_INET6, host)
+        host = f"[{host}]"
+
+    protocol = "http" if not https else "https"
+    url = f"{protocol}://{host}:5601/auth/login"
+    arguments = {"url": url}
+    if https:
+        arguments["verify"] = "./ca.pem"
+
+    try:
+        response = requests.get(**arguments)
```

Comment:
Only connection errors? Why not SSL or HTTP errors as well?

Reply:
If it's an SSL error, then the dasboard is not unvailable, but has TLS misconfigured.
I can add a `return False` on that.
We do not `raise_for_status)` here, so I don't expect any HTTP errors.

Reply:
(NOTE: this function is meant to verify if the dashboard as a service is down. Misconfigured state doesn't count as down for us in this context. I can clarify in the docstring.)

Reply:
Check my comment above. Having the dashboard healthy is the "normal" scenario. Therefore, this method will largely be used as a double negation (i.e. I want to know when `not dashboard_unavailable()`).

In the case we invert to `dashboard_available()`, then it makes sense to catch all the `request` exceptions and return them as False, then returning:
`return 200 <= response.status and response.status < 300`

Reply:
Currently, this method will return: `True, False, throw exceptions`. It means a simple check will always have to be covered with a `try/except` block unnecessarily...

Reply:
No double negation. That would be a mis-use of this helper function. (I can clarify this in the docstring.)

We already have `access_dashboard()` and `access_all_dashboards()` functions used all across all tests (including the test module in and new test this PR). That is the one to test Dashboards full healthy availability. Should return `True` if Dashboads is in a perfect shape. `False` on partial availabilty.

The new method is to check that Dashboard is specifically down (fully unavailable), and should not return `True` otherwise. If the Dashboard is available to "any extent" it should be `False`.

I can clarify on docstings or naming terminology. But I what I have in mind is 2 very simple use `True`/`False` cases with no additional logic, answering these questions:
 1. "Is the Dashboards is up and all well?" (If anything is wrong, "up but not well", etc -> `False`)
 3. "Is Dasboards sevice totally down?" (If serivce is green/yellow/red, i.e. up to ANY extent -> `False`)

It may worth to highlight that Dashboards is an extremely senitive service, going down on any unpreferred condition :-)
IF Opensearch can't be contacted (let's say `kibanaserver` pw is wrong): Dashboards simpy does not come up. IT's completely unavailable. No "red" status. You just can't access the API at all.

This is the reason why I think it makes sense to have a specific test helper function dedicated to detect this particular scenario. 

NOTE: Exceptions shouldn't be thrown. Only `True`/`False`. Which exceptions do you expect to he thrown? I may have overlooked something. If there's any, that must be handled.

May this context better highlight the intention here? If yes,